ModuleSourceRemote String now considers query string (#31250)

fatahfattah opened the pull request:
This fixes an issue in the String() for ModuleSourceRemote in which it does not consider query strings (if present). Before this fix, any subdirectory would simply be appended to the ModuleSourceRemote.Package instead of inserting it between the root package and any query parameters.


Related: https://github.com/hashicorp/terraform/issues/31250

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/terraform?pullRequest=31636)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/terraform?pullRequest=31636)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/terraform?pullRequest=31636).</sub>

crw commented:
Thanks for this submission. Although I cannot commit to having this PR reviewed at this time, we acknowledge your contribution and appreciate it!

crw commented:
Hi @fatahfattah, if you could please write one or more tests for this feature (even simple tests will be fine), we can get this reviewed. Thanks again for your submission!

fatahfattah commented:
Hi @crw I added a few tests, please let me know if they are okay or if you require some changes. I noticed there was no separate testing suite for parsing the ModuleSourceRemote so I added that as well.

crw commented:
The team member who is going to review this is on vacation, but we'll make sure to review this as soon as he is back. Thanks!

apparentlymart commented:
Thanks for working on this @fatahfattah!

This implementation looks good to me, and so I'm going to merge it now.
